invenio_records_resources/services/custom_fields/text.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Custom Fields for InvenioRDM."""
+
+from marshmallow_utils.fields import SanitizedUnicode
+
+from .base import BaseCF
+
+
+class KeywordCF(BaseCF):
+    """Keyword custom field."""
+
+    def __init__(self, name, **kwargs):
+        """Constructor."""
+        super().__init__(name, **kwargs)
+
+    @property
+    def mapping(self):
+        """Return the mapping."""
+        return {"type": "keyword"}
+
+    @property
+    def field(self):
+        """Marshmallow field custom fields."""
+        return SanitizedUnicode(**self._field_args)
+
+
+class TextCF(KeywordCF):
+    """Text custom field."""
+
+    def __init__(self, name, use_as_filter=False, **kwargs):
+        """Constructor."""
+        self.use_as_filter = use_as_filter
+        super().__init__(name, **kwargs)
+
+    @property
+    def mapping(self):
+        """Return the mapping."""
+        _mapping = {"type": "text"}
+        if self.use_as_filter:
```

Comment:
why not always adding the `keyword` by default?
There is also this [ignore_above](https://www.elastic.co/guide/en/elasticsearch/reference/current/ignore-above.html) which might be useful...

Reply:
Not in every case we want a keyword. But is true that we might want to add is as default (and deactivate if not needed). I'm adding this to the issue for OS compatibility, since this shoudl be subtituted by classes.